### PR TITLE
Constraint names

### LINKED ISSUE #1842: Need to document upcoming default naming scheme for constraints
As we are changing the constraint names (cf. https://github.com/prisma/prisma/issues/6680), we need to document the upgrade path for people to cope with the change.

1. Migrate your database so that the underlying constraints follow the new naming scheme
OR 
2. Introspect your database so that your Prisma schema reflects the database's underlying constraint names that are already there, thus preventing migrate to generate migrations for these in the future

### LINKED ISSUE #1869: Constraint names
Required steps:

- [ ] Replace 2.XX.0 (search and replace) with whatever the version is when this is released

### PR DESCRIPTION
**Pull request by mhwelander**
Before merging: 

* [ ] Address remaining conversations
* [ ] Address to-dos in #1869 issue

### CODE REVIEW

**Review by janpio — CHANGES_REQUESTED** [review]
Just to make sure this does not accidentally get merged by anyone - the feature is not ready yet.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 1]
```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
...
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
+
+2. Choose an [an upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
```
Comment (mhwelander) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
```suggestion
2. Choose an [upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
```

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+<TopBlock>
+
+This page describes manual upgrade steps that you need to perform after [upgrading to Prisma 2.27.0](.) if you want to use the `namedConstraints` Preview feature.
+
+</TopBlock>
+
+## Optional: Enable the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+If you choose to enable the `namedConstraints` Preview feature, choose an workflow (Prisma Migrate or Introspection) and upgrade path that suits your requirements.
```
Comment (mhwelander) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]:
```suggestion
If you choose to enable the `namedConstraints` Preview feature, choose a workflow (Prisma Migrate or Introspection) and upgrade path that suits your requirements.
```

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2-27-0.mdx [hunk 3]
```diff
@@ -1,9 +1,145 @@
...
+
+1. Introspect to populate your Prisma schema with any constraint names that do not follow Prisma's naming convention:
+
+   ```terminal
+   npx prisma introspect
+   ```
+
+   For example, in the following model, the `@id` primary key constraint does not follow Prisma's convention (the conventional name would be `User_id_pk`):
+
+   model User {
```
Comment (molebox) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2-27-0.mdx]: I think this little block is missing some back ticks 😄

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 4]
```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
...
+Named constraints is a Preview feature. To enable this feature:
+
+1. Add `namedConstraints` to the `previewFeatures` field in the `generator` block:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["namedConstraints"]
+}
+```
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
````suggestion
	  ```prisma
	  generator client {
	    provider        = "prisma-client-js"
	    previewFeatures = ["namedConstraints"]
	  }
	  ```
````

Indent so this is in the list

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 5]
```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
 
 **Relational database connectors** implement `autoincrement()`, `dbgenerated()`, and `now()` at database level. The **MongoDB connector** does not support `autoincrement()`, and `now()` is implemented at Prisma level. `dbgenerated()` can only be used to generate an `ObjectId`.
 
-## Relations
+## Constraints and indexes in the Prisma schema
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: This headline is weird as it says "in the Prisma Schema" although we are in a "Prisma schema" docs page. 
Without it, it is "too random" though as well.

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Unique and Index are both attributes, maybe this is more about "Naming Attributes"? 
Update: Well, except for Relations which are its own thing of course and destroy this idea :(

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Another thought: This section somehow feels like it belong into a "SQL concepts in Prisma schema" page or section.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 6]
```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
...
+
+- `@unique` represents a unique constraint on a single column
+- `@@index` represets an index on one or more columns
+- `@relation` represents a foreign key constraint
+
+Constraints and indexes have **names** in the underlying database. You can [define constraint and index names in the Prisma schema in 2.27.0 and later](#constraint-and-index-names-preview). 
+
+Foreign key constraints - also called referential constraints - also define a referential **action**. You can [define referential actions in the Prisma schema in 2.26.0 and later](#referential-actions-preview).
+
+### Constraint and index names (Preview)
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: This could be its own preview feature page until we go GA and embed the information above with the actual attributes?

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+
+### Prisma Migrate workflow
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]: I fear this does not say much for our users. We use that internally as "a thing", but users might not?

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+
+Use the following guide if you use Prisma Migrate. Choose the upgrade path that matches your requirements.
+
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx:23]: List of headlines with links here to make it easier to get an overview of the available options

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+   npx prisma introspect
+   ```
+
+   For example, in the following model, the `@id` primary key constraint does not follow Prisma's convention (the conventional name would be `User_id_pk`):
+
+   model User {
+     id    Int    @id(map: "user_id_primary_key") @default(autoincrement())
+     name  String @unique
+     posts Post[]
+   }
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]:
````suggestion
   ```prisma
   model User {
     id    Int    @id(map: "user_id_primary_key") @default(autoincrement())
     name  String @unique
     posts Post[]
   }
   ```   
````

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 10]
```diff
@@ -1010,9 +994,72 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
...
+
+2. Choose an [an upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+   <Admonition type="alert">
+
+   If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#prisma-migrate-workflow).
+   
+   </Admonition>
+
+In the following schema defines two constraints that follow convention, and one constraint and an index that do not:
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: I felt the example was a bit hard to follow, as I needed to parse the example to figure out which is which. Maybe the same schema once with default, and once with customized instead?

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx (hunk 7 above)
Comment (mhwelander) [reply content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]: I changed it to 'Upgrade paths for Prisma Migrate users' and 'Upgrade path for introspection only users'

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 5 above)
Comment (mhwelander) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: As per discussion - there is now a 'naming stuff' page under Prisma schema

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 6 above)
Comment (mhwelander) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Now on its own naming page so does not take up as much space - linked from preview page

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 10 above)
Comment (mhwelander) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Expanded - now with constraints/indexes before and after custom names

File: 2 [hunk 11]
```diff
@@ -0,0 +1 @@
+Terminate batch job (Y/N)?
```
Comment (Jolg42) [inline 2]: This is weird

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 12]
```diff
@@ -283,30 +285,12 @@ model Tag {
...
-  @@map(name: "comments")
-}
-```
-
-With this model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
+<Admonition type="info">
 
-> **Note**: You can also [`@map`](../../../reference/api-reference/prisma-schema-reference#map) <span class="api"></span> a column name or enum value, and `@@map` an enum.
+**Table or collection names in the underlying database** <br />
+You can optionally [map Prisma model names to a different collection / table name](names-in-underlying-database#mapping-collectiontable-and-fieldcolumn-names) in the underlying database.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
```suggestion
You can optionally [map Prisma model names to a different table/collection name](names-in-underlying-database#mapping-collectiontable-and-fieldcolumn-names) in the underlying database.
```

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 13]
```diff
@@ -1010,9 +1007,17 @@ Support for [`autoincrement()`](../../../reference/api-reference/prisma-schema-r
 
 **Relational database connectors** implement `autoincrement()`, `dbgenerated()`, and `now()` at database level. The **MongoDB connector** does not support `autoincrement()`, and `now()` is implemented at Prisma level. `dbgenerated()` can only be used to generate an `ObjectId`.
 
-## Relations
+<!-- ## Constraints and indexes in the Prisma schema
 
-Refer to the [relations documentation](relations) for more examples and information about relationships between models.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: This removal intended?

File: content/200-concepts/100-components/250-preview-features/050-client-preview-features.mdx [hunk 14]
```diff
@@ -11,6 +11,7 @@ The following [Preview](../../../../about/releases#preview) feature flags are av
 * [`selectRelationCount`](../prisma-client/aggregation-grouping-summarizing#count-relations) [Submit feedback](https://github.com/prisma/prisma/issues/6312)
 * [`orderByAggregateGroup`](../prisma-client/aggregation-grouping-summarizing#order-by-aggregate-group-preview) [Submit feedback](https://github.com/prisma/prisma/issues/6545)
 * [`referentialActions`](../prisma-schema/relations/referential-actions)
```
Comment (janpio) [inline content/200-concepts/100-components/250-preview-features/050-client-preview-features.mdx]:
```suggestion
* [`referentialActions`](../prisma-schema/relations/referential-actions) [Submit feedback](https://github.com/prisma/prisma/issues/7816)
```

File: content/200-concepts/100-components/250-preview-features/050-client-preview-features.mdx [hunk 15]
```diff
@@ -11,6 +11,7 @@ The following [Preview](../../../../about/releases#preview) feature flags are av
...
+* [`namedConstraints`](../prisma-schema/names-in-underlying-database#constraint-and-index-names-preview)
```
Comment (janpio) [inline content/200-concepts/100-components/250-preview-features/050-client-preview-features.mdx:14]:
```suggestion
* [`namedConstraints`](../prisma-schema/names-in-underlying-database#constraint-and-index-names-preview)  [Submit feedback](TODO)
```

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 16]
```diff
@@ -1033,12 +1033,15 @@ Defines a single-field ID on the model.
 
 #### Arguments
 
-N/A
+| Name     | Required | Type               | Description                                                                 |
+| -------- | -------- | ------------------ | --------------------------------------------------------------------------- |
+| `map` ([2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later)   | No       | `String`           | The constraint name in the underlying database. Default naming convention: `{table_name}_pk` (for example, `users_pk`)    |
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: Technically only with the preview feature flag and then later GA

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 17]
```diff
@@ -1280,9 +1283,11 @@ Defines a multi-field ID (composite ID) on the model.
 
 #### Arguments
 
-| Name     | Required | Type               | Description                                                      |
-| -------- | -------- | ------------------ | ---------------------------------------------------------------- |
-| `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]` |
+| Name     | Required | Type               | Description                                                                 |
+| -------- | -------- | ------------------ | --------------------------------------------------------------------------- |
+| `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]`            |
+| `name`   | No       | `String`           | The name of the field in the generated client - defaults to `field1_field2` |
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: Is `name` new for `@@id` @do4gr?

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 18]
```diff
@@ -2347,12 +2355,13 @@ Defines meta information about the relation. [Learn more](../../concepts/compone
...
 | `references` | `FieldReference[]` | On [annotated](../../../../../../concepts/components/prisma-schema/relations#annotated-relation-fields-and-relation-scalar-fields) relation fields | A list of [fields](../../../concepts/components/prisma-schema/data-model#defining-fields of the model on _the other side of the relation_ | `["id"]`, `["firstName, lastName"]`                   |
+| `map` ([2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later)  | No   | `String`             | The index name in the underlying database. Default naming convention: `{table_name}_{column_names}_fkey` (for example, `post_userid_fkey`)|
 
 The name of the `name` argument on the `@relation` attribute can be omitted (`references` is required):
 
 ```prisma
 @relation(name: "UserOnPost", references: [id])
-@relation("UserOnPost", references: [id])
+@relation("UserOnPost", references: [id], map: "")
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: Is empty `map` possible?

File: content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+
+For example, when you introspect a database with a table named `comments`, the result Prisma model will look like this:
+
+```prisma
+model comments {
+   // Fields
+}
+```
+
+However, you can still adhere to the naming convention without renaming the underlying `comments` table in the database by using the [`@@map`](../../../reference/api-reference/prisma-schema-reference#map-1) <span class="api"></span> attribute:
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx]:
```suggestion
However, you can still follow the naming convention without renaming the underlying `comments` table in the database by using the [`@@map`](../../../reference/api-reference/prisma-schema-reference#map-1) <span class="api"></span> attribute:
```
"adhere to" sounds like it is a must

File: content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+
+```prisma
+model Comment {
+   // Fields
+  @@map(name: "comments")
+}
+```
+
+With this model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx]:
```suggestion
With this modified model definition, Prisma automatically maps the `Comment` model to the `comments` table in the underlying database.
```

File: content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx [hunk 21]
```diff
@@ -0,0 +1,148 @@
...
+  @@map(name: "comments")
+}
+
+enum Type {
+  Blog,
+  Twitter @map("comment_twitter")
+  
+  @@map("comment_source_enum")
+}
+```
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx]: This is a bit confusing - maybe split them apart so it is easier to parse.

File: content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+
+`@map` and `@@map` allow you to [tune the shape of your Prisma Client API](../prisma-client/working-with-prismaclient/use-custom-model-and-field-names#using-map-and-map-to-rename-fields-and-models-in-the-prisma-client-api) by decoupling model and field names from table and column names in the underlying database.
+
+## Constraint and index names (Preview)
+
+In [2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later, you can use the `map` field to define the **underlying constraint and index names** in the Prisma schema. The `map` field is _only_ visible in the schema if you use a constraint name that differs from Prisma's naming convention.
+
+<Admonition type="warning">
+
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx:69]:
```suggestion
<Admonition type="warning">
```

File: content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+    provider        = "prisma-client-js"
+    previewFeatures = ["namedConstraints"]
+  }
+  ```
+
+2. Choose an [upgrade path](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#optional-enable-the-namedconstraints-preview-feature) depending on whether you use the Prisma Migrate or Introspection workflow.
+
+   <Admonition type="alert">
+
+   If you use Prisma Migrate and want to maintain your existing constraint and index names, **do not** enable the `namedConstraints` feature and immediately run `prisma migrate dev`. This will **change any underlying constraint name that does not follow Prisma's convention**. Follow the [upgrade path for the Prisma Migrate workflow](../../../guides/upgrade-guides/upgrading-to-latest/upgrade-to-2270#prisma-migrate-workflow).
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx:93]: 👍

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 24]
```diff
@@ -0,0 +1,150 @@
...
+## Optional: Enable the <inlinecode>namedConstraints</inlinecode> Preview feature
+
+If you choose to enable the `namedConstraints` Preview feature, choose a workflow (Prisma Migrate or Introspection) and upgrade path that suits your requirements:
+
+### Upgrade paths for Prisma Migrate users
+
+Use the following guide if your project uses Prisma Migrate. Choose the upgrade path that matches your requirements:
+
+* [I want to maintain my existing constraint and index names](#upgrade-path-1-i-want-to-maintain-my-existing-constraint-and-index-names)
+* [I want to use Prisma's default constraint and index namess](#upgrade-path-2-i-want-to-use-prismas-default-constraint-and-index-namess)
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]:
```suggestion
* [I want to use Prisma's default constraint and index names](#upgrade-path-2-i-want-to-use-prismas-default-constraint-and-index-names)
```

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 25]
```diff
@@ -0,0 +1,150 @@
...
+   
+2. Introspect your database to populate the Prisma schema with constraint and index names in your underlying database _that do not match Prisma's naming convention_:
+
+   ```terminal
+   npx prisma migrate introspect
+   ```
+
+   In this example, the highlighted constraints did not conform to Prisma's default naming convention and now include the `map` attribute field:
+
+   ```prisma
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]:
```suggestion
   ```prisma highlight=2,11;normal
```

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 26]
```diff
@@ -0,0 +1,150 @@
...
+   
+   ```prisma highlight=5;delete|6;add
+   model Post {
+     id    Int @id @default(autoincrement())
+     title String
+     authorName String @default("Anonymous")
+     author User? @relation(fields: [authorName], references: [name], map: "Custom_Foreign_Key_Constraint")
+     author User? @relation(fields: [authorName], references: [name])
+   }
+   ```
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]: They could modify the introspected values to match whatever they now want?
Maybe these three (currently 2) steps can be rolled into one?

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx [hunk 27]
```diff
@@ -0,0 +1,150 @@
...
+    }s
+    ```
+
+1. Run the `prisma migrate dev` command to generate a new migration:
+
+   ```terminal
+   npx prisma migrate dev
+   ```
+
+   This migration renames any constraints that do not follow Prisma's naming convention.
```
Comment (janpio) [inline content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]:
```suggestion
   This migration renames any constraints that do not currently follow Prisma's naming convention.
```

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 28]
```diff
@@ -2347,12 +2355,13 @@ Defines meta information about the relation. [Learn more](../../concepts/compone
...
 | `fields`     | `FieldReference[]` | On [annotated](../../../../../../concepts/components/prisma-schema/relations#annotated-relation-fields-and-relation-scalar-fields) relation fields | A list of [fields](../../../concepts/components/prisma-schema/data-model#defining-fields of the _current_ model                           | `["authorId"]`, `["authorFirstName, authorLastName"]` |
 | `references` | `FieldReference[]` | On [annotated](../../../../../../concepts/components/prisma-schema/relations#annotated-relation-fields-and-relation-scalar-fields) relation fields | A list of [fields](../../../concepts/components/prisma-schema/data-model#defining-fields of the model on _the other side of the relation_ | `["id"]`, `["firstName, lastName"]`                   |
+| `map` ([2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later)  | No   | `String`             | The index name in the underlying database. Default naming convention: `{table_name}_{column_names}_fkey` (for example, `post_userid_fkey`)|
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: This table is currently broken in the preview 😿

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 29]
```diff
@@ -1485,9 +1490,9 @@ Defines a [default value for a field](../../../concepts/components/prisma-schema
 
 #### Arguments
 
-| Name         | Required | Type                                      | Description |
-| ------------ | -------- | ----------------------------------------- | ----------- |
-| `expression` | **Yes**  | An expression (e.g. `5`, `true`, `now()`) |             |
+| Name         | Required | Type                                      | Description                                                              |
+| ------------ | -------- | ----------------------------------------- | ------------------------------------------------------------------------ |
+| `expression` | **Yes**  | An expression (e.g. `5`, `true`, `now()`) |                                                                          |
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: This is missing `map` for now (which might be SQL Server only, so not sure how that works yet)

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 30]
```diff
@@ -2246,19 +2254,19 @@ Manually defining an `@@index` in the Prisma schema serves no purpose.
...
 | `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]`                                                                                                                                                                            |
-| `name`   | **No**   | `name`             | The name of the index in the underlying database (Prisma generates an index name that respects identifier length limits if you do not specify a name. Prisma uses the following naming convention: `tablename.field1_field2_field3_unique`) |
+| `map` ([2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later)  | No   | `String`             | The index name in the underlying database. Default naming convention: `{table_name}_{column_names}_idx` (for example, `users_col1_col2_idx`) for globally unique indexes and `{column_names}_idx` for indexes that are only unique at table level (for example: `col1_idx1`) |
 
 The _name_ of the `fields` argument on the `@@index` attribute can be omitted:
 
 ```prisma no-lines
 @@index(fields: [title, author])
-@@index([title, author])
+@@index([title, author], map: "MyConstraintName")
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:2218]: This makes it look a bit like the omitting was possible because the `map` was added - maybe a separate example for `map`?

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx [hunk 31]
```diff
@@ -2010,20 +2017,21 @@ Defines a compound [unique constraint](../../../concepts/components/prisma-schem
...
+| `name`   | **No**   | `String`           | The name of the unique combination of fields that is [reflected in the generated Prisma Client](#specify-a-custom-name-for-a-multi-field-unique-attribute) - defaults to `fieldName1_fieldName2_fieldName3`             |
+| `map` ([2.27.0](https://github.com/prisma/prisma/releases/tag/2.27.0) and later)  | No   | `String`             | The constraint name in the underlying database. Default naming convention: `{table_name}_{column_names}_uniq` (for example, `users_col1_col2_uniq`) |
 
 The name of the `fields` argument on the `@@unique` attribute can be omitted:
 
 ```prisma no-lines
 @@unique(fields: [title, author])
 @@unique([title, author])
-@@unique(fields: [title, author], name: "titleAuthor")
+@@unique(fields: [title, author], name: "titleAuthor", map: "myConstraintName")
```
Comment (janpio) [inline content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:1987]: This makes it look a bit like the omitting was possible because the `map` was added - maybe a separate example for `map`?

File: content/400-reference/200-api-reference/100-prisma-schema-reference.mdx (hunk 17 above)
Comment (mhwelander) [reply content/400-reference/200-api-reference/100-prisma-schema-reference.mdx]: "@@id(map: database_pk_name) ← gets name = expose to Client (if missing = still constructed from field names)"

File: content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx (hunk 26 above)
Comment (mhwelander) [reply content/300-guides/300-upgrade-guides/200-upgrading-to-latest/10-upgrade-to-2270.mdx]: I guess so - I separated them so that you can address each choice separately (why would I remove, why would I rename)

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 13 above)
Comment (mhwelander) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Yep I moved the section

File: content/100-getting-started/02-setup-prisma/100-start-from-scratch/120-mongodb/100-connect-your-database.mdx [hunk 32]
```diff
@@ -0,0 +1,88 @@
...
+
+The format of the connection URL for your database depends on the database you use. For MongoDB, it looks as follows (the parts spelled all-uppercased are _placeholders_ for your specific connection details):
+
+```no-lines
+mongodb://USERNAME:PASSWORD@HOST:PORT/DATABASE
+```
+
+Here's a short explanation of each component:
+
+- `USERNAME`: The name of your database user
```
Comment (rahul3v) [inline content/100-getting-started/02-setup-prisma/100-start-from-scratch/120-mongodb/100-connect-your-database.mdx:46]:
```suggestion
- `USERNAME`: The username of your database user
```

File: content/100-getting-started/index.mdx [hunk 33]
```diff
@@ -24,16 +24,24 @@ The **easiest** way to get up and running with Prisma is through our [Quickstart
     Quickstart
 </ButtonLink>
 
-If you're looking to incorporate Prisma with an **existing project**, then the aptly named [Add to existing project](./setup-prisma/add-to-existing-project-typescript-postgres) guide will show you how.
+If you're looking to incorporate Prisma with an **existing project**, choose whether you want to use a SQL database or MongoDB:
```
Comment (rahul3v) [inline content/100-getting-started/index.mdx:27]:
```suggestion
If you're looking to incorporate Prisma with an **existing project**, choose whether you want to use a SQL database (like PostgreSQL, MySQL) or MongoDB (a NoSQL database):
```

File: content/100-getting-started/index.mdx [hunk 34]
```diff
@@ -24,16 +24,24 @@ The **easiest** way to get up and running with Prisma is through our [Quickstart
...
 
-To learn how to **create a project** with Prisma from scratch, the [Start from scratch](./setup-prisma/start-from-scratch-typescript-postgres) section guides you through creating a project, connecting your database and querying your data.
+<ButtonLink color="dark" type="primary" href="./getting-started/setup-prisma/add-to-existing-project/mongodb-typescript-mongodb">
+    Existing MongoDB project
+</ButtonLink>
+
+To learn how to **create a project** with Prisma from scratch, choose between SQL and MongoDB. Both tutorials will guide you through creating a project, connecting your database, and querying your data.
+
+<ButtonLink color="dark" type="primary" href="./getting-started/setup-prisma/start-from-scratch/relational-databases-typescript-postgres">
+    New SQL project
```
Comment (rahul3v) [inline content/100-getting-started/index.mdx:40]:
```suggestion
    Start new SQL project
```

File: content/100-getting-started/index.mdx [hunk 35]
```diff
@@ -24,16 +24,24 @@ The **easiest** way to get up and running with Prisma is through our [Quickstart
...
+</ButtonLink>
 
-<ButtonLink color="dark" type="primary" href="./getting-started/setup-prisma/start-from-scratch-typescript-postgres">
-    Start from scratch
+<ButtonLink color="dark" type="primary" href="./getting-started/setup-prisma/start-from-scratch/mongodb-typescript-mongodb">
+    New MongoDB project
```
Comment (rahul3v) [inline content/100-getting-started/index.mdx:44]:
```suggestion
    Start new MongoDB project
```

**Comment by janpio** [comment]
Thanks for the reviews @rahul3v, but that is probably more something to do via a new PR - as this one will be have to be rebased and probably be recreated against `main` before it can and will be merged - none of these changes should be in here.

**Comment by rahul3v** [comment]
Well @janpio , I guess I have to re create the PR once merged,  with all this changes then :)
or you can make it as you say !

File: content/200-concepts/100-components/01-prisma-schema/06-relations/index.mdx [hunk 36]
```diff
@@ -369,707 +356,182 @@ model Post {
...
+- you define a [self-relation](self-relations)
+- you define [an m-n for MongoDB](many-to-many-relations#mongodb)
+- you need to control how the relation table is represented in the underlying database (e.g. use a specific name for a relation table)
 
-This relation expresses the following:
+[Implicit m-n relations](many-to-many-relations#implicit-many-to-many-relations) in relational databases do not require the `@relation` attribute.
 
-- "a user has zero or one _teachers_ "
-- "a user can have zero or more _students_"
+<Admonition type="info">
```
Comment (do4gr) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/index.mdx:479]: to port
